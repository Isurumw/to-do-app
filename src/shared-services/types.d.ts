declare interface IStorage {
  get(key: string): Promise<string | undefined>;
  getObject<P>(key: string): Promise<P | undefined>;
  set(key: string, value: string): Promise<void>;
  remove(key: string): void;
}

declare interface INetworkManager {
  request(
    route: string,
    method: 'POST' | 'GET' | 'PUT' | 'DELETE' | 'PATCH',
    variables?: object | null,
  ): any;
}
