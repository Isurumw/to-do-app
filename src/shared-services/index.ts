import Config from 'react-native-config';
import AsyncStorage from '@react-native-community/async-storage';

import _NetworkManager from './network-services/network-manager';
import _StorageService from './storage-service/storage-service';

const StorageService = new _StorageService(AsyncStorage);
const NetworkManager = new _NetworkManager(
  Config.AUTH_BASE,
  Config.AUTH_KEY,
  StorageService,
  fetch,
);
const FirestoreManager = new _NetworkManager(
  Config.FIRESTORE_BASE,
  Config.AUTH_KEY,
  StorageService,
  fetch,
);

export {NetworkManager, FirestoreManager, StorageService};
