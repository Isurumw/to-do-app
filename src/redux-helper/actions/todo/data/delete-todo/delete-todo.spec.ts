import {FirestoreManager} from 'shared-services';
import deleteTodo from './delete-todo';

jest.mock('shared-services/network-services/network-manager');
const mockedFirestoreManager =
  FirestoreManager as unknown as jest.Mocked<INetworkManager>;

const mockName =
  'projects/todo-8013f/databases/(default)/documents/todos/FDv0SIXHmpCCcHX3Cp7S';

describe('check the behaviour of the delete-todo natwork service', () => {
  afterEach(() => {
    mockedFirestoreManager.request.mockClear();
  });

  it('deleted todos successful', async () => {
    mockedFirestoreManager.request.mockImplementation(() => {});

    await deleteTodo(mockName);

    expect(mockedFirestoreManager.request).toBeCalledTimes(1);
  });

  it('delete todos failed with an error', async () => {
    mockedFirestoreManager.request.mockImplementation(() => {
      throw new Error('Some API Error');
    });

    expect(deleteTodo(mockName)).rejects.toThrowErrorMatchingSnapshot();
  });
});
