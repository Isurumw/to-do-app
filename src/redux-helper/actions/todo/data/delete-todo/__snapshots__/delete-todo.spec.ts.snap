// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check the behaviour of the delete-todo natwork service delete todos failed with an error 1`] = `"Some API Error"`;
